x-pack/plugins/ml/server/routes/management.ts

```diff
@@ -0,0 +1,149 @@
+                };
+              });
+
+              return response.ok({
+                body: adJobsWithSpaces,
+              });
+            case 'data-frame-analytics':
+              const [
+                { data_frame_analytics: dfaJobs },
+                { data_frame_analytics: dfaJobsStats },
+                dfaJobStatus,
+              ] = await Promise.all([
+                mlClient.getDataFrameAnalytics({
+                  size: 10000,
+                }),
+                mlClient.getDataFrameAnalyticsStats({
+                  size: 10000,
+                }),
+                jobsSpaces(),
+              ]);
+
+              const dfaStatsMapped = dfaJobsStats.reduce((acc, cur) => {
+                acc[cur.id] = cur;
+                return acc;
+              }, {} as Record<string, estypes.MlDataframeAnalytics>);
+
+              const dfaJobsWithSpaces: AnalyticsManagementItems[] = dfaJobs.map((j) => {
+                const id = j.id;
+                return {
+                  id,
+                  description: j.description ?? '',
+                  source_index: j.source.index as string[], // esclient types are wrong
+                  dest_index: j.dest.index,
+                  job_type: Object.keys(j.analysis)[0] ?? '',
+                  state: dfaStatsMapped[id]?.state ?? '',
+                  spaces: dfaJobStatus['data-frame-analytics'][id] ?? [],
+                };
+              });
+              return response.ok({
+                body: dfaJobsWithSpaces,
+              });
+
+            case 'trained-model':
+              const [
+                { trained_model_configs: models },
+                { trained_model_stats: modelsStats },
+                modelSpaces,
+              ] = await Promise.all([
+                mlClient.getTrainedModels(),
+                mlClient.getTrainedModelsStats(),
+                trainedModelsSpaces(),
+              ]);
+
+              const modelStatsMapped = modelsStats.reduce((acc, cur) => {
+                acc[cur.model_id] = cur;
+                return acc;
+              }, {} as Record<string, estypes.MlTrainedModelStats>);
+
+              const modelsWithSpaces: TrainedModelsManagementItems[] = models.map((m) => {
+                const id = m.model_id;
+                return {
+                  id,
+                  description: m.description ?? '',
+                  state: modelStatsMapped[id].deployment_stats?.state ?? '',
+                  type: [
+                    m.model_type,
+                    ...Object.keys(m.inference_config),
+                    ...(m.tags.includes(BUILT_IN_MODEL_TAG) ? [BUILT_IN_MODEL_TYPE] : []),
+                  ],
+                  spaces: modelSpaces.trainedModels[id] ?? [],
+                };
+              });
+              return response.ok({
+                body: modelsWithSpaces,
+              });
+            default:
+              throw Error('aagghhh');
```

Comment:
We probably should change this error message

Reply:
Changed in https://github.com/elastic/kibana/pull/134893/commits/0c08ba996272059b4ad78b8e9cfd4fbb1b0735bd